api/src/endpoints/habitats.ts

```diff
@@ -0,0 +1,40 @@
+
+import { query } from '../db'
+import { parseArgs } from './habitatsArgParser'
+
+/*
+    example: POST /habitats
+    {
+        "framework": "liveng0"
+    }
+*/
+
+export const getHabitats = async (args: any) => {
+
+    console.log(`At ${(new Date()).toISOString()} - entering habitats function`)
+
+    let q = parseArgs(args)    
+
+    let habitats = await getHabitatsImpl(q)
+    
+    console.log(`At ${(new Date()).toISOString()} - got habitats query result`)
+
+    return { habitats }
+}
+
+export let getHabitatsImpl = async (q: { framework: string }) => {
+    
+    let sql = `
+      select
+         id,
+         habitat
+      from habitats
+      where framework = $1
+      order by habitat
+      limit 999
```

Comment:
What if there are more then 999 habitat types, have you checked the current frameworks? If it is an issue you could page with limit and offset